frontend/src/common/components/NavBar/Notification/AppointmentNotification.tsx

```diff
@@ -0,0 +1,194 @@
+import { useEffect, useState } from "react"
+import {
+  AppointmentNotificationType,
+  AppointmentNotificationTypes
+} from "../../../types/NotificationTypes"
+import { NotificationService } from "../../../services/NotificationService"
+import { specialityMapper } from "../../../../pages/FortuneTellerDetailPage/components/Packages/PackageList"
+import { Speciality } from "../../../../pages/FortuneTellerDetailPage/types/PackageTypes"
+import ReportModal from "../../../../pages/ConversationPage/components/ReportModal"
+
+const typeMapper: Record<AppointmentNotificationType, string> = {
+  NEW: "ต้องการนัดหมาย",
+  ACCEPT: "ได้ตอบรับการนัดหมาย",
+  DENY: "ได้ปฏิเสธการนัดหมาย",
+  CANCEL: "ได้ยกเลิกการนัดหมาย",
+  REMINDER: "",
+  COMPLETE: ""
+}
+
+export default function AppointmentNotification({
+  notificationId,
+  userId
+}: {
+  notificationId: string
+  userId: string
+}) {
+  const [appointmentNotification, setAppointmentNotification] =
+    useState<AppointmentNotificationTypes>()
+  const [isShowReport, setIsShowReport] = useState(false)
+
+  function translateAppointmentNotificationType(type: AppointmentNotificationType) {
+    return typeMapper[type]
+  }
+
+  function translateSpeciality(specialty: Speciality) {
+    return specialityMapper[specialty]
+  }
+
+  function addTimes(date: Date, minutes: number, hours: number): Date {
+    const result = new Date(date)
+    result.setMinutes(result.getMinutes() + minutes)
+    result.setHours(result.getHours() + hours)
+    return result
+  }
+
+  function padTo2Digits(num: number): string {
+    return num.toString().padStart(2, "0")
+  }
+
+  function showDate(date: Date): string {
+    date = addTimes(date, 0, 7)
+    return (
+      padTo2Digits(date.getDate()) +
+      "/" +
+      padTo2Digits(date.getMonth() + 1) +
+      "/" +
+      date.getFullYear()
+    )
+  }
+
+  function showTime(date: Date, duration: number): string {
+    date = addTimes(date, duration, 7)
+    return padTo2Digits(date.getHours()) + "." + padTo2Digits(date.getMinutes())
+  }
+
+  const showReport = () => {
+    setIsShowReport(true)
+  }
+
+  useEffect(() => {
+    const fetchAppointmentNotification = async ({
+      notificationId,
+      userId
+    }: {
+      notificationId: string
+      userId: string
+    }) => {
+      const appointmentNotification = await NotificationService.getAppointmentNotification(
+        notificationId,
+        userId
+      )
+      setAppointmentNotification(appointmentNotification)
+    }
+    fetchAppointmentNotification({ notificationId, userId })
+  }, [])
+
+  if (appointmentNotification)
```

Comment:
function นี้อาจจะไม่ return อะไรออกไปเลยอ่ะ ซึ่งอาจจะพังได้ เข้าใจว่าพึ่ง fetch data ใน level นี้

มีสองทางที่แนะนำคือ
1. สร้าง type NONE ของ Appointment มา แล้ว init state ด้วย NONE ละ return <></>

2. แยกการ fetch กับ ui เป็น 2 layers
  fetch layer ก็คือจะ fetch อย่างเดียว ถ้าไม่เจอก็ handle ไป
แต่ถ้าเจอก็ส่งให้ ui layer ซึ่ง ui layer จะการันตีมีข้อมูลให้แสดง แล้ว จะไม่มีพวก type undefined เกิดขึ้น อันนี้ clean สุด

3. (Not recommended) 
if(!appointmentNotification){
    return
}